tests/Unit/ApparentHorizons/Test_Strahlkorper.cpp

```diff
@@ -0,0 +1,206 @@
+    }
+  }
+
+  for (size_t l = sk.ylm_spherepack().l_max() + 1;
+       l <= sk_high_res.ylm_spherepack().l_max(); ++l) {
+    for (int m = -static_cast<int>(l); m <= static_cast<int>(l); ++m) {
+      CHECK(final_coefs[iter_high_res.set(l, m)()] == approx(0.0));
+    }
+  }
+}
+
+void test_average_radius() {
+  const std::array<double, 3> center = {{0.1, 0.2, 0.3}};
+  const double r = 3.0;
+  Strahlkorper<Frame::Inertial> s(4, 4, r, center);
+  CHECK(s.average_radius() == approx(r));
+}
+
+void test_copy_and_move() {
+  Strahlkorper<Frame::Inertial> s(4, 4, 3.0, {{0.1, 0.2, 0.3}});
+
+  test_copy_semantics(s);
+  auto s_copy = s;
+  test_move_semantics(std::move(s), s_copy);
+}
+
+void test_physical_center() {
+  const std::array<double, 3> physical_center = {{1.5, 0.5, 1.0}};
+  const std::array<double, 3> expansion_center = {{0.0, 0.0, 0.0}};
+  const double radius = 5.0;
+  const int l_max = 9;
+
+  Strahlkorper<Frame::Inertial> sk(l_max, l_max, radius, expansion_center);
+  DataVector r(sk.ylm_spherepack().physical_size(), 0.);
+
+  for (size_t s = 0; s < r.size(); ++s) {
+    const double theta = sk.ylm_spherepack().theta_phi_points()[0][s];
+    const double phi = sk.ylm_spherepack().theta_phi_points()[1][s];
+    // Compute the distance (radius as a function of theta,phi) from
+    // the expansion_center to a spherical surface of radius `radius`
+    // centered at physical_center.
+    const double a = 1.0;
+    const double b = -2.0 * cos(phi) * sin(theta) * physical_center[0] -
+                     2.0 * sin(phi) * sin(theta) * physical_center[1] -
+                     2.0 * cos(theta) * physical_center[2];
+    const double c = physical_center[0] * physical_center[0] +
+                     physical_center[1] * physical_center[1] +
+                     physical_center[2] * physical_center[2] - radius * radius;
+    auto roots = real_roots(a, b, c);
+    r[s] = std::max(roots[0], roots[1]);
+  }
+  // Construct a new Strahlkorper sk_test with the radius computed
+  // above, centered at expansion_center, so that
+  // sk_test.physical_center() should recover the physical center of
+  // this surface.
+  Strahlkorper<Frame::Inertial> sk_test(l_max, l_max, r, expansion_center);
+  for (size_t i = 0; i < 3; ++i) {
+    CHECK(approx(gsl::at(physical_center, i)) ==
+          gsl::at(sk_test.physical_center(), i));
+  }
+}
+
+void test_point_is_contained() {
+  // Construct a spherical Strahlkorper
+  const double radius = 2.;
+  const std::array<double, 3> center = {{-1.2, 3., 4.}};
+  const Strahlkorper<Frame::Inertial> sphere(3, 2, radius, center);
+
+  // Check whether two known points are contained.
+  const std::array<double, 3> point_inside = {{-1.2, 1.01, 4.}};
+  const std::array<double, 3> point_outside = {{-1.2, 3., 6.01}};
+  CHECK(sphere.point_is_contained(point_inside));
+  CHECK(not sphere.point_is_contained(point_outside));
+}
+
+void test_constructor_with_different_coefs() {
```

Comment:
It looks like the three versions are identical except the constructor call of `test2`. Could you just pass in a lambda that does a different constructor call to reduce this to one function?